curriculum/tools/utils.js

```diff
@@ -0,0 +1,182 @@
+const fs = require('fs');
+const path = require('path');
+const matter = require('gray-matter');
+
+const padWithLeadingZeros = originalNum => {
+  /* always want file step numbers 3 digits */
+  const maxDigits = 3;
+  let paddedNum = '' + originalNum;
+  while (paddedNum.length < maxDigits) {
+    paddedNum = '0' + paddedNum;
+  }
+  return paddedNum;
+};
+
+const removeErms = seedCode => {
+  const lines = seedCode.split('\n');
+  return lines
+    .filter(line => !line.includes('--fcc-editable-region--'))
+    .join('\n');
+};
+
+/*
+function below modifed from GitHub user @solenoid's gist https://gist.github.com/solenoid/1372386
+*/
+const mongoObjectId = () => {
+  /* eslint-disable no-bitwise */
+  var timestamp = ((new Date().getTime() / 1000) | 0).toString(16);
+  return (
+    timestamp +
+    'xxxxxxxxxxxxxxxx'
+      .replace(/[x]/g, function() {
+        return ((Math.random() * 16) | 0).toString(16);
+      })
+      .toLowerCase()
+  );
+};
```

Comment:
This function does seem to generate valid ids, at least according to the Mongo driver and Mongoose. I'm just wondering if it would be better to use the Mongo package to generate the ids.

Seems like Mongo itself uses BSON to generate the ids, so that might be a lightweight alternative. There's also the [bson-objectid](https://www.npmjs.com/package/bson-objectid) package, which is even lighter than the regular BSON one.

Either way, this seems to work fine and shouldn't cause collisions.

Reply:
I'd rather not reinvent things, as a rule. If that bson package seems fine, can we use that?